pkg/setting/setting.go

```diff
@@ -341,8 +341,10 @@ type Cfg struct {
 
 	ApiKeyMaxSecondsToLive int64
 
-	// Use to enable new features which may still be in alpha/beta stage.
-	FeatureToggles       map[string]bool
+	// Check if a feature toggle is enabled
+	// @deprecated
+	IsFeatureToggleEnabled func(key string) bool // filled in dynamically
```

Comment:
The circular dependencies make this pretty hard to get rid of :grimacing: 

Most places that used the map reference can be replaced with a refernce to `featuremgmt.FeatureToggles`, but a few places can not.

In particular the `settings.Provider` can not use featuremgmt without a circular ref